### PR TITLE
Spelling compiler

### PR DESCRIPTION
**Pull request by jsoref**
This is per https://github.com/rust-lang/rust/pull/110392#issuecomment-1510193656

I'm going to delay performing a squash because I really don't expect people to be perfectly happy w/ my changes, I really am a human and I really do make mistakes.

r? Nilstrieb

I'm going to be flying this evening, but I should be able to squash / respond to reviews w/in a day or two.

I tried to be careful about dropping changes to `tests`, afaict only two files had changes that were likely related to the changes for a given commit (this is where not having eagerly squashed should have given me an advantage), but, that said, picking things apart can be error prone.

### CODE REVIEW

File: compiler/rustc_feature/src/removed.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ declare_features! (
     (removed, const_compare_raw_pointers, "1.46.0", Some(53020), None,
      Some("cannot be allowed in const eval in any meaningful way")),
     /// Allows non-trivial generic constants which have to be manually propagated upwards.
-     (removed, const_evaluatable_checked, "1.48.0", Some(76560), None, Some("renamed to `generic_const_exprs`")),
+     (removed, const_evaluable_checked, "1.48.0", Some(76560), None, Some("renamed to `generic_const_exprs`")),
```
Comment (jsoref) [inline compiler/rustc_feature/src/removed.rs:63]: It's possible this token is tied to an older version of rust and thus not changeable. Dunno...

File: compiler/rustc_feature/src/removed.rs [hunk 2]
```diff
@@ -139,7 +139,7 @@ declare_features! (
     /// Allows using `#[on_unimplemented(..)]` on traits.
     /// (Moved to `rustc_attrs`.)
     (removed, on_unimplemented, "1.40.0", None, None, None),
-    /// A way to temporarily opt out of opt in copy. This will *never* be accepted.
+    /// A way to temporarily opt out of opt-in copy. This will *never* be accepted.
```
Comment (jsoref) [inline compiler/rustc_feature/src/removed.rs:142]: Here's a case with both a verb phrase (`opt out`) and a noun (`opt-in`) where the noun has compounded/hyphenated (`-`).

File: compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs [hunk 3]
```diff
@@ -794,7 +794,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             return;
         };
 
-        // get all where BoundPredicates here, because they are used in to cases below
+        // get all where BoundPredicates here, because they are used in two cases below
```
Comment (jsoref) [inline compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs:797]: This is an atypical correction (I don't generally check 2-letter words), the `in to` check flagged this. I'm pretty sure my correction is right.

File: compiler/rustc_incremental/src/persist/dirty_clean.rs [hunk 4]
```diff
@@ -139,7 +139,7 @@ pub fn check_dirty_clean_annotations(tcx: TyCtxt<'_>) {
         return;
     }
 
-    // can't add `#[rustc_clean]` etc without opting in to this feature
+    // can't add `#[rustc_clean]` etc without opting into this feature
```
Comment (jsoref) [inline compiler/rustc_incremental/src/persist/dirty_clean.rs:142]: `opt*in*to` is complicated, there are definitely strong differences of opinion on it.

**Comment by Nilstrieb** [comment]
I will review this once CI passes. Looks like some functions got renamed but not everywhere.
@rustbot author

File: compiler/rustc_infer/src/infer/canonical/query_response.rs [hunk 5]
```diff
@@ -467,11 +467,11 @@ impl<'tcx> InferCtxt<'tcx> {
                     }
                 }
                 GenericArgKind::Const(result_value) => {
-                    if let ty::ConstKind::Bound(debrujin, b) = result_value.kind() {
+                    if let ty::ConstKind::Bound(debruijn, b) = result_value.kind() {
```
Comment (jsoref) [inline compiler/rustc_infer/src/infer/canonical/query_response.rs:470]: Both spellings were used, this appears to both be dominant and to match the wider world: https://en.wikipedia.org/wiki/De_Bruijn_sequence
(It's also a thing I got to learn a little about by virtue of reading this repository's code...)

File: compiler/rustc_mir_build/src/build/matches/test.rs [hunk 6]
```diff
@@ -499,7 +499,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
     /// However, in some cases, the test may just not be relevant to candidate.
     /// For example, suppose we are testing whether `foo.x == 22`, but in one
     /// match arm we have `Foo { x: _, ... }`... in that case, the test for
-    /// what value `x` has has no particular relevance to this candidate. In
+    /// what value `x` has no particular relevance to this candidate. In
```
Comment (jsoref) [inline compiler/rustc_mir_build/src/build/matches/test.rs:502]: this isn't technically a spelling error, but spell checkers have been flagging doubled words for decades.
sometimes doubled words are intentional -- this doesn't appear to be one such case.

File: compiler/rustc_apfloat/src/ieee.rs [hunk 7]
```diff
@@ -2645,7 +2645,7 @@ mod sig {
 
         // Convert the result having "2 * precision" significant-bits back to the one
         // having "precision" significant-bits. First, move the radix point from
-        // poision "2*precision - 1" to "precision - 1". The exponent need to be
+        // poisson "2*precision - 1" to "precision - 1". The exponent need to be
```
Comment (Nilstrieb) [inline compiler/rustc_apfloat/src/ieee.rs:2648]: as mentioned previously, all changes in `rustc_apfloat` need to be undone, as that crate should not be touched due to being in licensing limbo

File: compiler/rustc_error_codes/src/error_codes/E0026.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-A struct pattern attempted to extract a non-existent field from a struct.
+A struct pattern attempted to extract a nonexistent field from a struct.
```
Comment (jsoref) [inline compiler/rustc_error_codes/src/error_codes/E0026.md:1]: This appears to relate to tests/ui/error-codes/E0026-teach.stderr

File: tests/ui/error-codes/E0026-teach.stderr
Comment (jsoref) [inline tests/ui/error-codes/E0026-teach.stderr:1]: This appears to relate to compiler/rustc_error_codes/src/error_codes/E0026.md

File: compiler/rustc_hir_analysis/src/astconv/mod.rs
Comment (jsoref) [inline compiler/rustc_hir_analysis/src/astconv/mod.rs:1]: This appears to relate to tests/ui/type-alias-enum-variants/enum-variant-generic-args.stderr

File: tests/ui/type-alias-enum-variants/enum-variant-generic-args.stderr
Comment (jsoref) [inline tests/ui/type-alias-enum-variants/enum-variant-generic-args.stderr:1]: This appears to relate to compiler/rustc_hir_analysis/src/astconv/mod.rs

**Comment by jsoref** [comment]
This `is_const_evaluatable` stuff appears to be a mess that also involves crossing rust-clippy (I have an open PR for stuff in rust-clippy that shouldn't do any cross stuff) (in my original work, I was pretty sure that CI was happy with me, but splitting is hazardous and I'm imperfect). I'm going to drop it and if/when this merges, I can approach it separately.

File: compiler/rustc_ast_passes/messages.ftl [hunk 9]
```diff
@@ -231,7 +231,7 @@ ast_passes_feature_on_non_nightly = `#![feature]` may not be used on the {$chann
     .suggestion = remove the attribute
     .stable_since = the feature `{$name}` has been stable since `{$since}` and no longer requires an attribute to enable
 
-ast_passes_incompatbile_features = `{$f1}` and `{$f2}` are incompatible, using them at the same time is not allowed
+ast_passes_incompatible_features = `{$f1}` and `{$f2}` are incompatible, using them at the same time is not allowed
```
[the remark this answers is not in the document]
Comment (Nilstrieb) [reply compiler/rustc_ast_passes/messages.ftl:234]: right now all translation keys are defined here in tree and only english is available so this is no concern

File: compiler/rustc_feature/src/removed.rs (hunk 1 above)
Comment (Nilstrieb) [reply compiler/rustc_feature/src/removed.rs]: this should indeed not be changed

**Comment by jsoref** [comment]
I've hit that ordering thing a couple of times and I thought I fixed it a couple of times. It's taken me a while to understand why it's happening (because I see the beginning word and treat it as if it's the same for both of the first two tokens, but, of course by fixing the spelling of the beginning word) -- the second word now sorts properly instead of jumping the queue because it was sorting against the last letter of the previous word.

File: compiler/rustc_mir_build/src/build/matches/test.rs (hunk 6 above)
Comment (Nilstrieb) [reply compiler/rustc_mir_build/src/build/matches/test.rs:502]: the double word is correct here (although it makes the sentence a little awkward) - feel free to either reword the sentence of drop the change

File: compiler/rustc_incremental/src/persist/dirty_clean.rs (hunk 4 above)
Comment (Nilstrieb) [reply compiler/rustc_incremental/src/persist/dirty_clean.rs:142]: i dont have an opinion here, applying the change is less work for us both

File: compiler/rustc_mir_build/src/build/matches/test.rs (hunk 6 above)
Comment (jsoref) [inline compiler/rustc_mir_build/src/build/matches/test.rs:502]: Ah indeed. Hmm. How about:
```suggestion
    /// the value of `x` has no particular relevance to this candidate. In
```

Comment (Nilstrieb) [reply compiler/rustc_mir_build/src/build/matches/test.rs:502]: sounds good

File: compiler/rustc_resolve/src/errors.rs [hunk 10]
```diff
@@ -22,7 +22,7 @@ pub(crate) struct UnderscoreLifetimeNameCannotBeUsedHere(#[primary_span] pub(cra
 
 #[derive(Diagnostic)]
 #[diag(resolve_crate_may_not_be_imported)]
-pub(crate) struct CrateMayNotBeImprted(#[primary_span] pub(crate) Span);
+pub(crate) struct CrateMayNotBeImported(#[primary_span] pub(crate) Span);
```
Comment (Nilstrieb) [inline compiler/rustc_resolve/src/errors.rs:25]: This struct is dead code, no one mentions it. You shouldn't remove it here, but this is interesting anyways. Looks like something from the diagnostics derive is messing up the dead code lint (cc @davidtwco).

**Review by Noratrieb — APPROVED** [review]
r=me after squashing

**Review by Noratrieb — APPROVED** [review]

**Pull request merged**